cpp/src/fil/fil.cu

```diff
@@ -35,35 +35,37 @@ creation and prediction (the main inference kernel is defined in infer.cu). */
 namespace ML {
 namespace fil {
 
+__host__ __device__ double sigmoid(double x) { return 1.0 / (1.0 + exp(-x)); }
```

Comment:
Any reason not to templatize these two functions?

Reply:
Done (templatized).